### PR TITLE
feat: add superhtml support

### PR DESCRIPTION
**Pull request by EliSauder**
Added support the superhtml lsp server.
[source](https://github.com/kristoff-it/superhtml/tree/main)
[article](https://handmade.network/p/564/superhtml-an-html-language-server/)

This is not quite ready to go since the lsp appears to not be functioning fully (that or I've set it up wrong) https://github.com/kristoff-it/superhtml/issues/20.

### CODE REVIEW

**Comment by EliSauder** [comment]
@glepnir this should be ready to go now. superhtml has been up versioned to v0.4.3 and the configuration now works. If you would like details I can provide them or you can find them in the link in the initial issue.

**Comment by EliSauder** [comment]
Some updates: the superhtml filetype has been added into neovim, patch: https://github.com/neovim/neovim/commit/e596b6a18d84aa56b5189bb7921feec58279976c

File: lua/lspconfig/server_configurations/superhtml.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'superhtml', 'lsp' },
+    filetypes = { 'superhtml', 'html' },
+    root_dir = util.root_pattern '.git',
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/superhtml.lua:7]:
```suggestion
    root_dir = util.find_git_ancestor
```

**Comment by EliSauder** [comment]
@glepnir with those changes am I good to remove the draft status?

**Pull request merged**